### PR TITLE
Implement fmt::format_to into std::vector<char>

### PR DESCRIPTION
**Pull request by phprus**
Fix for issue #4206

### CODE REVIEW

File: include/fmt/base.h [hunk 1]
```diff
@@ -1971,12 +1971,32 @@ template <typename T = char> class counting_buffer : public buffer<T> {
...
+          FMT_ENABLE_IF(has_container_append<T, It>::value)>
+FMT_CONSTEXPR20 void container_append(T& container, It begin, It end) {
+  container.append(begin, end);
+}
+
+template <typename T, typename It,
+          FMT_ENABLE_IF(!has_container_append<T, It>::value)>
+FMT_CONSTEXPR20 void container_append(T& container, It begin, It end) {
+  container.insert(container.end(), begin, end);
+}
```
Comment (vitaut) [inline include/fmt/base.h:1993]: I don't think we need an extra "level of indirection". Let's merge these into `copy`.

**Review by vitaut — CHANGES_REQUESTED** [review]
Thanks for the PR! Overall looks good, but let's simplify this a bit (see the inline comment).

File: include/fmt/base.h (hunk 1 above)
Comment (phprus) [reply include/fmt/base.h]: Merged

File: include/fmt/base.h [hunk 2]
```diff
@@ -1971,15 +1971,38 @@ template <typename T = char> class counting_buffer : public buffer<T> {
...
+    : std::true_type {};
+
 // An optimized version of std::copy with the output value type (T).
-template <typename T, typename InputIt, typename OutputIt,
-          FMT_ENABLE_IF(is_back_insert_iterator<OutputIt>::value)>
+template <
+    typename T, typename InputIt, typename OutputIt,
+    FMT_ENABLE_IF(
+        is_back_insert_iterator<OutputIt>::value&& has_container_append<
+            decltype(get_container(std::declval<OutputIt>())), InputIt>::value)>
```
[the remark this answers is not in the document]
Comment (phprus) [reply include/fmt/base.h]: Done

**Review by vitaut — APPROVED** [review]

**Comment by vitaut** [comment]
Thanks!

**Pull request merged**